From 817c5ca575de3323d7f17b373ae11119dfee5341 Mon Sep 17 00:00:00 2001
From: Anurag <anurag.gopinath@broadcom.com>
Date: Fri, 11 Jan 2013 17:52:20 +0530
Subject: [PATCH 659/762] MMC : Moving number of slot information to DTS files

Removing board specific hardcode for 'number of MMC slots'
and moving 'number of MMC slots' information to DTS files.

Based on Broadcom SDK 2.3.

Signed-off-by: Anurag <anurag.gopinath@broadcom.com>
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/mmc/host/sdhci-xlp.c |   17 ++++++++++-------
 1 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/drivers/mmc/host/sdhci-xlp.c b/drivers/mmc/host/sdhci-xlp.c
index 6aa598b..b510851 100644
--- a/drivers/mmc/host/sdhci-xlp.c
+++ b/drivers/mmc/host/sdhci-xlp.c
@@ -35,6 +35,7 @@
 #include <asm/netlogic/hal/nlm_hal_xlp_dev.h>
 
 #include "sdhci.h"
+#include "../boot/ops.h"
 #include "sdhci-pltfm.h"
 
 /*****************************************************************************\
@@ -59,12 +60,12 @@ struct sdhci_xlp_node {
 
 
 static int findmaxslots(void) {
-        if(is_nlm_xlp8xx())
-                return XLP_SINGLE_SD_SLOT;
-        else if (is_nlm_xlp3xx())
-                return XLP_DOUBLE_SD_SLOT;
-        else
-                return XLP_SINGLE_SD_SLOT;
+        int slots = 1;
+        void *node;
+        node = finddevice("/doms/dom@0/mmc");
+        if(node) 
+            getprop(node, "slots", &slots, 4);
+        return slots;    
 }
 
 /*****************************************************************************\
@@ -200,7 +201,7 @@ static int __devinit sdhci_pltfm_probe(struct platform_device *pdev)
                 goto err;
         }
         mmc_ctrl_base = sys_addr;
-        mmc_ctrl_base[0>>2] = 0x1C;;
+        mmc_ctrl_base[0>>2] = 0x14;
 
         /*
         * The XLP MMC/SD controller has two slots. The registers for the
@@ -211,6 +212,8 @@ static int __devinit sdhci_pltfm_probe(struct platform_device *pdev)
         slots = findmaxslots();
 
         for (slotno = 0; slotno < slots; slotno++) {
+            if(slotno)
+                mmc_ctrl_base[0>>2] = mmc_ctrl_base[0>>2]| 0x8;
             pdev->resource[slotno].flags = IORESOURCE_MEM;
             pdev->resource[slotno].start = iomem->start + (0x100); 
             pdev->resource[slotno].end = pdev->resource[slotno].start + 
-- 
1.7.0.4

